parley/src/main/java/nu/parley/android/DefaultParleyLaunchCallback.java

```diff
@@ -0,0 +1,36 @@
+package nu.parley.android;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+
+import androidx.annotation.NonNull;
+import androidx.core.app.ActivityCompat;
+
+public class DefaultParleyLaunchCallback implements ParleyLaunchCallback {
+
+    private final Context context;
```

Comment:
No that would be unnecessary, for the following reason:

The `DefaultParleyLaunchCallback` is applied to, and held in memory, by a View. It uses the View Context. When the Context is no longer valid (not referenced because the Activity is destroyed) any Views part of that Activity will also no longer be referenced, this also leads to this listener no longer being referenced*

_*Actually it is referenced, but only by other objects which themselves are no longer referenced, such as the View. Meaning GC will mark everything as 'collectable'._

Only if something keeps referencing the `DefaultParleyLaunchCallback` while the Context is no longer valid, then it would be leaking the Context. However in that case I would argue that it is not the fault of `DefaultParleyLaunchCallback`, instead it is the fault of the application using this callback. The reasoning behind this is that it is well understood that when something takes Context as input, you should not be leaking that object beyond the Contexts lifecycle. Examples of this include Views, they take Context, and it is well known to not leak those. 

In my opinion `WeakReferences` should be a last resort, when you can't properly control the lifecycle of an object in relation to the Context that it holds. In this case the lifecycle of `DefaultParleyLaunchCallback` is really well managed (it is only referenced in the View) and if anyone wants to reference it somewhere where it might leak, then they might want to resort to use `WeakReference<DefaultParleyLaunchCallback>`, but even better de-reference it when the Context is no longer valid.

Reply:
Thanks for the detailed reasoning, agreed! 👍 

Also since `DefaultParleyLaunchCallback` will most likely not be used by others, it's good like this as long as we handle it correct inside Parley.